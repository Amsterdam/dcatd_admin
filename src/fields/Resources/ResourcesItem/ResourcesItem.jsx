import React from 'react';
import PropTypes from 'prop-types';

import filesize from 'filesize';
import dateFormat from '../../../definitions/date-format';

import './resources-item.scss';

function getFileType(mime, fieldSchema) {
  const type = fieldSchema.enum.indexOf(mime) > -1 ? fieldSchema.enumNames[fieldSchema.enum.indexOf(mime)].toLowerCase() : 'overig';
  return (
    <span className={`resources-item__file-type
      resources-item__file-type--${type}`}
    >{type}
    </span>
  );
}

const ResourcesItem = ({ resource, schemaProps }) => (
  <div className="resources-item">
    <div className="resources-item__modified">
      {dateFormat.formatDate(resource['dct:modified'])}</div>
    <div className="resources-item__file-size">
      {resource['dcat:byteSize'] > 0 ? filesize(resource['dcat:byteSize']) : ''}</div>
    <div className="resources-item__title">
      {resource['dct:title']}</div>
    <div className="resources-item__description">
      { (resource['ams:distributionType'] !== 'web') && getFileType(resource['dct:format'], schemaProps['dct:format'])}
      <span className="resources-item__description-text">{resource['dct:description'] || resource['dcat:accessURL']}</span>
    </div>
  </div>
);

ResourcesItem.defaultProps = {
  resource: {}
};

ResourcesItem.propTypes = {
  resource: PropTypes.object,
  schemaProps: PropTypes.object.isRequired
};

export default ResourcesItem;
