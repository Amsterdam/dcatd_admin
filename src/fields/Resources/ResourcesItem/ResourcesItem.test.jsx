import React from 'react';

import ResourcesItem from './ResourcesItem';

const mockSchema = {
  'dct:format': {
    enum: [
      'n/a',
      'text/csv',
      'application/vnd.openxmlformats-officedocument.wordprocessingml.document',
      'application/vnd.geo+json',
      'application/gml+xml',
      'text/html',
      'application/json',
      'application/pdf',
      'image/png',
      'application/vnd.ms-excel',
      'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
      'application/xml',
      'application/octet-stream'
    ],
    enumNames: ['', 'CSV', 'DOCX', 'GeoJSON', 'GML', 'HTML', 'JSON', 'PDF', 'PNG', 'XLS', 'XLSX', 'XML', 'Anders']
  }
};

describe('The ResourcesItem component', () => {
  it('renders with title, description, format, file size and date', () => {
    const resource = {
      'dct:title': 'Titel',
      'dcat:accessURL': 'http://ergens',
      'dct:description': 'omschrijving',
      'dct:format': 'application/pdf',
      'dcat:byteSize': '666',
      'foaf:isPrimaryTopicOf': {
        'dct:modified': '2017-11-30'
      }
    };

    const wrap = shallow(
      <ResourcesItem
        resource={resource}
        schemaProps={mockSchema}
      />
    );

    expect(wrap).toMatchSnapshot();
  });

  it('renders with title, description and date: leaving out format and file size for websites', () => {
    const resource = {
      'ams:distributionType': 'web',
      'dct:title': 'Titel',
      'dcat:accessURL': 'http://ergens',
      'dct:description': 'omschrijving',
      'dct:format': 'n/a',
      'dcat:byteSize': '0',
      'foaf:isPrimaryTopicOf': {}
    };

    const wrap = shallow(
      <ResourcesItem
        resource={resource}
        schemaProps={mockSchema}
      />
    );

    expect(wrap).toMatchSnapshot();
  });

  it('renders with title and url: showing url in place of description', () => {
    const resource = {
      'dct:title': 'Titel',
      'dcat:accessURL': 'http://ergens',
      'dct:format': 'n/a',
      'foaf:isPrimaryTopicOf': {}
    };

    const wrap = shallow(
      <ResourcesItem
        resource={resource}
        schemaProps={mockSchema}
      />
    );

    expect(wrap).toMatchSnapshot();
  });
});
