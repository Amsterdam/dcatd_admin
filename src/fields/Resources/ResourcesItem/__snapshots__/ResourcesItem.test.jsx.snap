// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The ResourcesItem component renders with title and url: showing url in place of description 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ResourcesItem
    resource={
      Object {
        "dcat:accessURL": "http://ergens",
        "dct:format": "n/a",
        "dct:title": "Titel",
        "foaf:isPrimaryTopicOf": Object {},
      }
    }
    schemaProps={
      Object {
        "dct:format": Object {
          "enum": Array [
            "n/a",
            "text/csv",
            "application/vnd.openxmlformats-officedocument.wordprocessingml.document",
            "application/vnd.geo+json",
            "application/gml+xml",
            "text/html",
            "application/json",
            "application/pdf",
            "image/png",
            "application/vnd.ms-excel",
            "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet",
            "application/xml",
            "application/octet-stream",
          ],
          "enumNames": Array [
            "",
            "CSV",
            "DOCX",
            "GeoJSON",
            "GML",
            "HTML",
            "JSON",
            "PDF",
            "PNG",
            "XLS",
            "XLSX",
            "XML",
            "Anders",
          ],
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="resources-item__modified"
        >
          
        </div>,
        <div
          className="resources-item__file-size"
        >
          
        </div>,
        <div
          className="resources-item__title"
        >
          Titel
        </div>,
        <div
          className="resources-item__description"
        >
          <span
            className="resources-item__file-type
      resources-item__file-type--"
          >
            
          </span>
          <span
            className="resources-item__description-text"
          >
            http://ergens
          </span>
        </div>,
      ],
      "className": "resources-item",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "className": "resources-item__modified",
        },
        "ref": null,
        "rendered": "",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "className": "resources-item__file-size",
        },
        "ref": null,
        "rendered": "",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Titel",
          "className": "resources-item__title",
        },
        "ref": null,
        "rendered": "Titel",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="resources-item__file-type
      resources-item__file-type--"
            >
              
            </span>,
            <span
              className="resources-item__description-text"
            >
              http://ergens
            </span>,
          ],
          "className": "resources-item__description",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
              "className": "resources-item__file-type
      resources-item__file-type--",
            },
            "ref": null,
            "rendered": "",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "http://ergens",
              "className": "resources-item__description-text",
            },
            "ref": null,
            "rendered": "http://ergens",
            "type": "span",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="resources-item__modified"
          >
            
          </div>,
          <div
            className="resources-item__file-size"
          >
            
          </div>,
          <div
            className="resources-item__title"
          >
            Titel
          </div>,
          <div
            className="resources-item__description"
          >
            <span
              className="resources-item__file-type
      resources-item__file-type--"
            >
              
            </span>
            <span
              className="resources-item__description-text"
            >
              http://ergens
            </span>
          </div>,
        ],
        "className": "resources-item",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "resources-item__modified",
          },
          "ref": null,
          "rendered": "",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "resources-item__file-size",
          },
          "ref": null,
          "rendered": "",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Titel",
            "className": "resources-item__title",
          },
          "ref": null,
          "rendered": "Titel",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="resources-item__file-type
      resources-item__file-type--"
              >
                
              </span>,
              <span
                className="resources-item__description-text"
              >
                http://ergens
              </span>,
            ],
            "className": "resources-item__description",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
                "className": "resources-item__file-type
      resources-item__file-type--",
              },
              "ref": null,
              "rendered": "",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "http://ergens",
                "className": "resources-item__description-text",
              },
              "ref": null,
              "rendered": "http://ergens",
              "type": "span",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`The ResourcesItem component renders with title, description and date: leaving out format and file size for websites 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ResourcesItem
    resource={
      Object {
        "ams:distributionType": "web",
        "dcat:accessURL": "http://ergens",
        "dcat:byteSize": "0",
        "dct:description": "omschrijving",
        "dct:format": "n/a",
        "dct:title": "Titel",
        "foaf:isPrimaryTopicOf": Object {},
      }
    }
    schemaProps={
      Object {
        "dct:format": Object {
          "enum": Array [
            "n/a",
            "text/csv",
            "application/vnd.openxmlformats-officedocument.wordprocessingml.document",
            "application/vnd.geo+json",
            "application/gml+xml",
            "text/html",
            "application/json",
            "application/pdf",
            "image/png",
            "application/vnd.ms-excel",
            "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet",
            "application/xml",
            "application/octet-stream",
          ],
          "enumNames": Array [
            "",
            "CSV",
            "DOCX",
            "GeoJSON",
            "GML",
            "HTML",
            "JSON",
            "PDF",
            "PNG",
            "XLS",
            "XLSX",
            "XML",
            "Anders",
          ],
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="resources-item__modified"
        >
          
        </div>,
        <div
          className="resources-item__file-size"
        >
          
        </div>,
        <div
          className="resources-item__title"
        >
          Titel
        </div>,
        <div
          className="resources-item__description"
        >
          <span
            className="resources-item__description-text"
          >
            omschrijving
          </span>
        </div>,
      ],
      "className": "resources-item",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "className": "resources-item__modified",
        },
        "ref": null,
        "rendered": "",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "className": "resources-item__file-size",
        },
        "ref": null,
        "rendered": "",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Titel",
          "className": "resources-item__title",
        },
        "ref": null,
        "rendered": "Titel",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            false,
            <span
              className="resources-item__description-text"
            >
              omschrijving
            </span>,
          ],
          "className": "resources-item__description",
        },
        "ref": null,
        "rendered": Array [
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "omschrijving",
              "className": "resources-item__description-text",
            },
            "ref": null,
            "rendered": "omschrijving",
            "type": "span",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="resources-item__modified"
          >
            
          </div>,
          <div
            className="resources-item__file-size"
          >
            
          </div>,
          <div
            className="resources-item__title"
          >
            Titel
          </div>,
          <div
            className="resources-item__description"
          >
            <span
              className="resources-item__description-text"
            >
              omschrijving
            </span>
          </div>,
        ],
        "className": "resources-item",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "resources-item__modified",
          },
          "ref": null,
          "rendered": "",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "resources-item__file-size",
          },
          "ref": null,
          "rendered": "",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Titel",
            "className": "resources-item__title",
          },
          "ref": null,
          "rendered": "Titel",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              false,
              <span
                className="resources-item__description-text"
              >
                omschrijving
              </span>,
            ],
            "className": "resources-item__description",
          },
          "ref": null,
          "rendered": Array [
            false,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "omschrijving",
                "className": "resources-item__description-text",
              },
              "ref": null,
              "rendered": "omschrijving",
              "type": "span",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`The ResourcesItem component renders with title, description, format, file size and date 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ResourcesItem
    resource={
      Object {
        "dcat:accessURL": "http://ergens",
        "dcat:byteSize": "666",
        "dct:description": "omschrijving",
        "dct:format": "application/pdf",
        "dct:title": "Titel",
        "foaf:isPrimaryTopicOf": Object {
          "dct:modified": "2017-11-30",
        },
      }
    }
    schemaProps={
      Object {
        "dct:format": Object {
          "enum": Array [
            "n/a",
            "text/csv",
            "application/vnd.openxmlformats-officedocument.wordprocessingml.document",
            "application/vnd.geo+json",
            "application/gml+xml",
            "text/html",
            "application/json",
            "application/pdf",
            "image/png",
            "application/vnd.ms-excel",
            "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet",
            "application/xml",
            "application/octet-stream",
          ],
          "enumNames": Array [
            "",
            "CSV",
            "DOCX",
            "GeoJSON",
            "GML",
            "HTML",
            "JSON",
            "PDF",
            "PNG",
            "XLS",
            "XLSX",
            "XML",
            "Anders",
          ],
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="resources-item__modified"
        >
          
        </div>,
        <div
          className="resources-item__file-size"
        >
          666 B
        </div>,
        <div
          className="resources-item__title"
        >
          Titel
        </div>,
        <div
          className="resources-item__description"
        >
          <span
            className="resources-item__file-type
      resources-item__file-type--pdf"
          >
            pdf
          </span>
          <span
            className="resources-item__description-text"
          >
            omschrijving
          </span>
        </div>,
      ],
      "className": "resources-item",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "className": "resources-item__modified",
        },
        "ref": null,
        "rendered": "",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "666 B",
          "className": "resources-item__file-size",
        },
        "ref": null,
        "rendered": "666 B",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Titel",
          "className": "resources-item__title",
        },
        "ref": null,
        "rendered": "Titel",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="resources-item__file-type
      resources-item__file-type--pdf"
            >
              pdf
            </span>,
            <span
              className="resources-item__description-text"
            >
              omschrijving
            </span>,
          ],
          "className": "resources-item__description",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "pdf",
              "className": "resources-item__file-type
      resources-item__file-type--pdf",
            },
            "ref": null,
            "rendered": "pdf",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "omschrijving",
              "className": "resources-item__description-text",
            },
            "ref": null,
            "rendered": "omschrijving",
            "type": "span",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="resources-item__modified"
          >
            
          </div>,
          <div
            className="resources-item__file-size"
          >
            666 B
          </div>,
          <div
            className="resources-item__title"
          >
            Titel
          </div>,
          <div
            className="resources-item__description"
          >
            <span
              className="resources-item__file-type
      resources-item__file-type--pdf"
            >
              pdf
            </span>
            <span
              className="resources-item__description-text"
            >
              omschrijving
            </span>
          </div>,
        ],
        "className": "resources-item",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "resources-item__modified",
          },
          "ref": null,
          "rendered": "",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "666 B",
            "className": "resources-item__file-size",
          },
          "ref": null,
          "rendered": "666 B",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Titel",
            "className": "resources-item__title",
          },
          "ref": null,
          "rendered": "Titel",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="resources-item__file-type
      resources-item__file-type--pdf"
              >
                pdf
              </span>,
              <span
                className="resources-item__description-text"
              >
                omschrijving
              </span>,
            ],
            "className": "resources-item__description",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "pdf",
                "className": "resources-item__file-type
      resources-item__file-type--pdf",
              },
              "ref": null,
              "rendered": "pdf",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "omschrijving",
                "className": "resources-item__description-text",
              },
              "ref": null,
              "rendered": "omschrijving",
              "type": "span",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
